

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Welcome to NiPTUNE’s documentation! &mdash; NiPTUNE  documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="#" class="icon icon-home" alt="Documentation Home"> NiPTUNE
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <!-- Local TOC -->
              <div class="local-toc"><ul>
<li><a class="reference internal" href="#">Welcome to NiPTUNE’s documentation!</a></li>
<li><a class="reference internal" href="#get-started">Get started</a></li>
<li><a class="reference internal" href="#installation">Installation</a></li>
<li><a class="reference internal" href="#dependencies">Dependencies</a></li>
<li><a class="reference internal" href="#configuration-file">Configuration File</a></li>
<li><a class="reference internal" href="#how-to-use-niptune">How to use NiPTUNE</a></li>
<li><a class="reference internal" href="#triton">Triton</a></li>
<li><a class="reference internal" href="#despina">Despina</a></li>
<li><a class="reference internal" href="#proteus">Proteus</a></li>
<li><a class="reference internal" href="#halimede">Halimede</a></li>
<li><a class="reference internal" href="#larissa">Larissa</a></li>
<li><a class="reference internal" href="#laomedeida">Laomedeida</a></li>
<li><a class="reference internal" href="#neso">Neso</a></li>
<li><a class="reference internal" href="#nereid">Nereid</a></li>
<li><a class="reference internal" href="#sao">Sao</a></li>
<li><a class="reference internal" href="#thalassa">Thalassa</a></li>
</ul>
</div>
            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="#">NiPTUNE</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="#" class="icon icon-home"></a> &raquo;</li>
        
      <li>Welcome to NiPTUNE’s documentation!</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/index.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="welcome-to-niptune-s-documentation">
<h1>Welcome to NiPTUNE’s documentation!<a class="headerlink" href="#welcome-to-niptune-s-documentation" title="Permalink to this headline">¶</a></h1>
<p>NiPTUNE is a python library giving tools to analyse NIPT data.
This library has been developed in python3.</p>
<div class="toctree-wrapper compound">
</div>
</div>
<div class="section" id="get-started">
<h1>Get started<a class="headerlink" href="#get-started" title="Permalink to this headline">¶</a></h1>
<p>Each module can be used as a python stand alone.</p>
</div>
<div class="section" id="installation">
<h1>Installation<a class="headerlink" href="#installation" title="Permalink to this headline">¶</a></h1>
<p>pip install -i <a class="reference external" href="https://test.pypi.org/simple/">https://test.pypi.org/simple/</a> NiPTUNE</p>
</div>
<div class="section" id="dependencies">
<h1>Dependencies<a class="headerlink" href="#dependencies" title="Permalink to this headline">¶</a></h1>
<ul>
<li><p>python 3.7:</p>
<blockquote>
<div><ul class="simple">
<li><p>matplotlib 3.3.2</p></li>
<li><p>numpy 1.19.2</p></li>
<li><p>pandas 1.1.3</p></li>
<li><p>pysam 0.16.0.1</p></li>
<li><p>statsmodels 0.12.0</p></li>
<li><p>scikit-learn 0.23.2</p></li>
</ul>
</div></blockquote>
</li>
<li><p>R 3.5.2:</p>
<blockquote>
<div><ul class="simple">
<li><p>ggfortify 0.4.11</p></li>
<li><p>ggplot2 3.3.2</p></li>
<li><p>DNAcopy 1.62.0</p></li>
<li><p>Rsamtools 2.4.0</p></li>
</ul>
</div></blockquote>
</li>
</ul>
</div>
<div class="section" id="configuration-file">
<h1>Configuration File<a class="headerlink" href="#configuration-file" title="Permalink to this headline">¶</a></h1>
<p>First you need to set the configuration file.
This file is already filled, we don’t recommend to modify the parameters, exept for the parameter [folders][home]</p>
<p><strong>Set path for NiPTUNE output</strong></p>
<p>In section [folders] the parameter <code class="docutils literal notranslate"><span class="pre">home</span></code> is set to niptune_results by default.
For every new analysis the user has to modify this parameter in order to not overwrite results already existing.</p>
<p><strong>Subdirectories of NiPTUNE output:</strong></p>
<ul>
<li><p>output_tables:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">output_dir</span><span class="o">/</span><span class="n">tables</span>
</pre></div>
</div>
</li>
</ul>
<blockquote>
<div><p>here you will find the outputs of despina.py, proteus.py, halimede.py, laomedeida.py, neso.py, nereid.py, sao.py that have a tables format</p>
</div></blockquote>
<ul>
<li><p>output_triton:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">output_dir</span><span class="o">/</span><span class="n">converted_files</span>
</pre></div>
</div>
<p>here you will find the converted files (.gcc, .pickle, .npz) from the module triton.py</p>
</li>
<li><p>output_proteus:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">output_dir</span><span class="o">/</span><span class="n">PCA_plots</span>
</pre></div>
</div>
<p>here you will find the PCA plots  from the module proteus.py</p>
</li>
<li><p>output_proteus_bin:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">output_dir</span><span class="o">/</span><span class="n">PCA_bins</span>
</pre></div>
</div>
<p>here you will find the distribution of reads count gc normalized by bin used as input fot PCA for the module proteus.py</p>
</li>
<li><p>output_halimede:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">output_dir</span><span class="o">/</span><span class="n">gender_prediction</span>
</pre></div>
</div>
<p>here you will find the threshold estimated by the gaussian mixture model from the module halimede.py to separate female/male populations.</p>
</li>
<li><p>output_halimede_Yfrac:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">output_dir</span><span class="o">/</span><span class="n">gender_prediction_MagicYcounts</span>
</pre></div>
</div>
<p>here you will find the estimated MagicY counts for each sample.</p>
</li>
<li><p>output_larissa:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">output_dir</span><span class="o">/</span><span class="n">reference</span>
</pre></div>
</div>
<p>here you will find the list of samples selected by larissa.py to be used as reference for defrag and WisecondorX</p>
</li>
<li><p>output_larissa_wisecondorX:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">output_dir</span><span class="o">/</span><span class="n">reference</span><span class="o">/</span><span class="n">npz</span>
</pre></div>
</div>
<p>here you will find the npz files (symbolic link to the folder output_triton) correspondent to the samples selected by larissa.py to be used as reference for WisecondorX</p>
</li>
<li><p>output_larissa_defrag:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">output_dir</span><span class="o">/</span><span class="n">reference</span><span class="o">/</span><span class="n">boydir</span> <span class="o">&amp;</span> <span class="n">output_dir</span><span class="o">/</span><span class="n">reference</span><span class="o">/</span><span class="n">girldir</span>
</pre></div>
</div>
<p>here you will find the gcc and pickle files (symbolic link to the folder output_triton) correspondent to the samples selected by larissa.py to be used as reference for Defrag</p>
</li>
<li><p>output_nereid:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">output_dir</span><span class="o">/</span><span class="n">read_len_dist</span>
</pre></div>
</div>
<p>here you will find the read length distribution for each sample</p>
</li>
<li><p>output_sao_data:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">output_dir</span><span class="o">/</span><span class="n">copy_number_alteration_data</span>
</pre></div>
</div>
<p>here you will find the bin visualization plots and the additional files created by WisecondorX  through sao.py.</p>
</li>
</ul>
</div>
<div class="section" id="how-to-use-niptune">
<h1>How to use NiPTUNE<a class="headerlink" href="#how-to-use-niptune" title="Permalink to this headline">¶</a></h1>
<p>First of all you need to create your configuration file in an empty directory.</p>
<p>Once it’s done go in directory where you created this file.
Then from here you can launch every NiPTUNE stand alone.</p>
<p>You can use NiPTUNE in 2 different ways : reference mode and routine mode.</p>
<dl class="simple">
<dt>If it’s the first time you are using NiPTUNE you need to create references for several tools:</dt><dd><ul class="simple">
<li><p>proteus</p></li>
<li><p>halimede</p></li>
<li><p>laomedeida</p></li>
<li><p>sao</p></li>
</ul>
</dd>
</dl>
<p>For the others tools there is no need of reference.</p>
<p>Usage example:</p>
<ul class="simple">
<li><p>Triton:</p></li>
</ul>
<blockquote>
<div><p>triton list_of_files.txt</p>
</div></blockquote>
<ul class="simple">
<li><p>Despina:</p></li>
</ul>
<blockquote>
<div><p>despina list_of_files.txt</p>
</div></blockquote>
<ul class="simple">
<li><p>Proteus:</p></li>
</ul>
<p>To create a reference:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">proteus</span> <span class="n">list_of_prefix</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<p>In routine</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">proteus</span> <span class="o">-</span><span class="n">r</span> <span class="n">list_of_prefix</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<ul class="simple">
<li><p>Halimede:</p></li>
</ul>
<p>To create a threshold of reference:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">halimede</span> <span class="o">-</span><span class="n">s</span> <span class="n">list_of_files</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<p>In routine</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">halimede</span> <span class="n">list_of_files</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<ul class="simple">
<li><p>Larissa:</p></li>
</ul>
<blockquote>
<div><p>larissa list_of_files.txt</p>
</div></blockquote>
<ul class="simple">
<li><p>Neso:</p></li>
</ul>
<blockquote>
<div><p>neso list_of_files.txt</p>
</div></blockquote>
<ul class="simple">
<li><p>Laomedeida:</p></li>
</ul>
<blockquote>
<div><p>laomedeida list_of_prefix.txt</p>
</div></blockquote>
<ul class="simple">
<li><p>Sao:</p></li>
</ul>
<blockquote>
<div><p>sao list_of_prefix.txt</p>
</div></blockquote>
<ul class="simple">
<li><p>Nereid:</p></li>
</ul>
<blockquote>
<div><p>nereid list_of_files.txt</p>
</div></blockquote>
<ul class="simple">
<li><p>Thalassa:</p></li>
</ul>
<blockquote>
<div><p>thalassa</p>
</div></blockquote>
</div>
<div class="section" id="triton">
<h1>Triton<a class="headerlink" href="#triton" title="Permalink to this headline">¶</a></h1>
<p>A package allowing to convert file from .bam format to other formats (.gcc, .pickle, .npz) which serve as input for downstream processing in NiPTUNE.
.gcc, .pickle and .npz are binned files.</p>
<p>Input: A list in .txt format containing list of samples in bam format. Indicate the absolute path of files.</p>
<p>For example see <code class="docutils literal notranslate"><span class="pre">tests/data/list_bam_files.txt</span></code></p>
<p>Output: Files are stored in the directory <code class="docutils literal notranslate"><span class="pre">converted_files</span></code></p>
<p>Arguments:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">-</span><span class="n">p</span><span class="p">,</span> <span class="o">--</span><span class="n">binSizePickle</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="mi">1</span><span class="n">Mb</span><span class="p">,</span> <span class="n">to</span> <span class="nb">set</span> <span class="n">the</span> <span class="n">size</span> <span class="n">of</span> <span class="n">bins</span> <span class="k">for</span> <span class="n">gcc</span> <span class="ow">and</span> <span class="n">pickle</span> <span class="n">files</span>
<span class="o">-</span><span class="n">n</span><span class="p">,</span> <span class="o">--</span><span class="n">binSizeNpz</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="mi">5</span><span class="n">Kb</span><span class="p">,</span> <span class="n">to</span> <span class="nb">set</span> <span class="n">the</span> <span class="n">size</span> <span class="n">of</span> <span class="n">bins</span> <span class="k">for</span> <span class="n">npz</span> <span class="n">files</span>
</pre></div>
</div>
</div>
<div class="section" id="despina">
<h1>Despina<a class="headerlink" href="#despina" title="Permalink to this headline">¶</a></h1>
<p>This module calculates the sequencing depth (sd) of the sample(s) provided.</p>
<p>Input : A list in .txt format containing list of samples in bam format. Indicate the absolute path of files.</p>
<p>For example see <code class="docutils literal notranslate"><span class="pre">tests/data/list_bam_files.txt</span></code></p>
<p>Output: one table with sample names and associated sd. Results are stored in the directory <code class="docutils literal notranslate"><span class="pre">tables</span></code></p>
<p>Arguments: none</p>
</div>
<div class="section" id="proteus">
<h1>Proteus<a class="headerlink" href="#proteus" title="Permalink to this headline">¶</a></h1>
<p>Principal component analysis. This module performs a principal component analysis on all the samples provided and produce a visual output to check whether there are outlier samples or subpopulations into the cohort.</p>
<p>Input: list of samples, only prefix file name (no path), previously converted in gcc format.</p>
<p>For example see <code class="docutils literal notranslate"><span class="pre">tests/data/list_prefix.txt</span></code></p>
<p>Output: one table with sample names and associated PCA plot. Results are stored in the directory <code class="docutils literal notranslate"><span class="pre">tables`.</span> <span class="pre">All</span> <span class="pre">PCA</span> <span class="pre">plots</span> <span class="pre">are</span> <span class="pre">stored</span> <span class="pre">in</span> <span class="pre">the</span> <span class="pre">directory</span> <span class="pre">``PCA_plots</span></code>. The distributions of reads count GC normalized by bin used as input for PCA for are stored in the directory <code class="docutils literal notranslate"><span class="pre">PCA_bins</span></code>.</p>
<p>Arguments::</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">-</span><span class="n">r</span><span class="p">,</span> <span class="o">--</span><span class="n">routine</span><span class="p">,</span> <span class="n">default</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span> <span class="n">to</span> <span class="n">choose</span> <span class="n">whether</span> <span class="n">you</span> <span class="n">need</span> <span class="n">to</span> <span class="n">use</span> <span class="n">PCA</span> <span class="ow">in</span> <span class="n">routine</span><span class="o">.</span> <span class="n">By</span> <span class="n">default</span> <span class="n">this</span> <span class="n">option</span> <span class="ow">is</span> <span class="nb">set</span> <span class="n">to</span> <span class="kc">False</span> <span class="ow">and</span> <span class="n">compute</span> <span class="n">a</span> <span class="n">PCA</span> <span class="n">of</span> <span class="n">reference</span> <span class="k">with</span> <span class="nb">all</span> <span class="n">the</span> <span class="n">cohort</span> <span class="n">data</span><span class="o">.</span>
</pre></div>
</div>
</div>
<div class="section" id="halimede">
<h1>Halimede<a class="headerlink" href="#halimede" title="Permalink to this headline">¶</a></h1>
<p>Fetal gender prediction with MagicY.</p>
<p>This module implements the MagicY method for fetal gender prediction. It is based on the quantification of reads in seven region on chrY and a gaussian mixture model to separate female and male population.</p>
<p>Input : A list in .txt format containing list of samples in bam format. Indicate the absolute path of files.</p>
<p>Output: one table with sample names and associated gender predicted (M/F). Results are stored in the directory <code class="docutils literal notranslate"><span class="pre">tables</span></code>. In its first application, MagicY needs to estimate a threshold to separate female and male population using a <strong>gaussian mixture model</strong>. This threshold is stored in the directory <code class="docutils literal notranslate"><span class="pre">gender_prediction</span></code>. In routine NIPT, you do not need to re-estimate this threshold, thus the only output will be the table stored in the directory <code class="docutils literal notranslate"><span class="pre">tables`.</span> <span class="pre">The</span> <span class="pre">estimated</span> <span class="pre">MagicY</span> <span class="pre">counts</span> <span class="pre">for</span> <span class="pre">each</span> <span class="pre">sample</span> <span class="pre">are</span> <span class="pre">stored</span> <span class="pre">in</span> <span class="pre">the</span> <span class="pre">directory</span> <span class="pre">``gender_prediction_MagicYcounts</span></code>.</p>
<p>Arguments:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">-</span><span class="n">s</span><span class="p">,</span> <span class="o">--</span><span class="n">setThreshold</span><span class="p">,</span> <span class="n">default</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span> <span class="n">to</span> <span class="n">choose</span> <span class="n">whether</span> <span class="n">you</span> <span class="n">need</span> <span class="n">to</span> <span class="n">calculate</span> <span class="n">the</span> <span class="n">threshold</span> <span class="ow">or</span> <span class="ow">not</span><span class="o">.</span>
<span class="o">-</span><span class="n">t</span><span class="p">,</span> <span class="o">--</span><span class="n">thresholdValue</span><span class="p">,</span> <span class="n">default</span> <span class="o">=</span> <span class="n">automatically</span> <span class="n">uses</span> <span class="n">the</span> <span class="n">threshold</span> <span class="n">estimated</span> <span class="n">when</span> <span class="n">the</span> <span class="n">option</span> <span class="n">s</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">to</span> <span class="nb">set</span> <span class="n">a</span> <span class="n">different</span> <span class="n">threshold</span> <span class="n">compared</span> <span class="n">to</span> <span class="n">the</span> <span class="n">one</span> <span class="n">estimated</span> <span class="n">by</span> <span class="n">the</span> <span class="n">gaussian</span> <span class="n">mixture</span> <span class="n">model</span> <span class="p">(</span><span class="ow">in</span> <span class="n">general</span> <span class="n">you</span> <span class="n">do</span> <span class="ow">not</span> <span class="n">need</span> <span class="n">to</span> <span class="nb">set</span> <span class="n">this</span> <span class="n">value</span><span class="p">,</span> <span class="n">we</span> <span class="n">suggest</span> <span class="n">to</span> <span class="n">use</span> <span class="n">this</span> <span class="n">option</span> <span class="n">only</span> <span class="k">for</span> <span class="n">testing</span> <span class="n">purposes</span><span class="p">)</span><span class="o">.</span>
</pre></div>
</div>
</div>
<div class="section" id="larissa">
<h1>Larissa<a class="headerlink" href="#larissa" title="Permalink to this headline">¶</a></h1>
<p>References creation for Defrag and WisecondorX. This module randomly selects n samples to be used as reference for Defrag and WisecondorX while keeping the gender ratio stable.</p>
<p>Input: list of samples, only prefix file name (no path), previously converted in gcc, pickle, npz formats and with gender predicted by MagicY</p>
<p>Output: list of samples selected as reference (stored in the directory reference), correspondent npz converted files (stored in the directory reference/npz), correspondent gcc and pickle converted files (stored in the directories reference/boydir and reference/girldir accordingly to sample gender).</p>
<p>Arguments:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">-</span><span class="n">n</span><span class="p">,</span> <span class="o">--</span><span class="n">numberSamples</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="mi">100</span><span class="p">,</span> <span class="n">to</span> <span class="nb">set</span> <span class="n">the</span> <span class="n">number</span> <span class="n">of</span> <span class="n">samples</span> <span class="n">to</span> <span class="n">create</span> <span class="n">reference</span><span class="o">.</span>
<span class="o">-</span><span class="n">b</span><span class="p">,</span> <span class="o">--</span><span class="n">binSizeNpz</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="mi">1</span><span class="n">Kb</span><span class="p">,</span> <span class="n">to</span> <span class="nb">set</span> <span class="n">the</span> <span class="n">size</span> <span class="n">of</span> <span class="n">bins</span> <span class="k">for</span> <span class="n">npz</span> <span class="n">conversion</span>
<span class="o">-</span><span class="n">c</span><span class="p">,</span> <span class="o">--</span><span class="n">cpus</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">to</span> <span class="nb">set</span> <span class="n">the</span> <span class="n">number</span> <span class="n">of</span> <span class="n">threads</span> <span class="n">to</span> <span class="n">use</span>
</pre></div>
</div>
</div>
<div class="section" id="laomedeida">
<h1>Laomedeida<a class="headerlink" href="#laomedeida" title="Permalink to this headline">¶</a></h1>
<p>ff estimation with Defrag a.</p>
<p>The main code of Defrag is maintained.</p>
<p>Input: list of samples, only prefix file name (no path); reference will be automatically found in reference/boydir and reference/girldir as determined by larissa.py.</p>
<p>Output: one table with sample names and associated ff estimated by Defrag a method. Results are stored in the <code class="docutils literal notranslate"><span class="pre">directory</span> <span class="pre">tables</span></code>.</p>
<p>Arguments:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">-</span><span class="n">g</span><span class="p">,</span> <span class="o">--</span><span class="n">GCcorrection</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">to</span> <span class="nb">set</span> <span class="n">whether</span> <span class="n">to</span> <span class="n">apply</span> <span class="n">GC</span> <span class="n">correction</span> <span class="n">before</span> <span class="n">Defrag</span> <span class="n">calculation</span><span class="o">.</span>
</pre></div>
</div>
</div>
<div class="section" id="neso">
<h1>Neso<a class="headerlink" href="#neso" title="Permalink to this headline">¶</a></h1>
<p>ff estimation with Seqff</p>
<p>Input: list of samples, only prefix file name (no path)</p>
<p>Output: one table with sample names and associated ff estimated by Seqff method. Results are stored in the <code class="docutils literal notranslate"><span class="pre">directory</span> <span class="pre">tables</span></code>.</p>
<p>Arguments: none</p>
</div>
<div class="section" id="nereid">
<h1>Nereid<a class="headerlink" href="#nereid" title="Permalink to this headline">¶</a></h1>
<p>Epsilon value estimation</p>
<p>Input : A list in .txt format containing list of samples in bam format. Indicate the absolute path of files.</p>
<p>Output: one table with sample names and associated e values for chr18 and chr21 each for ff estimated by Defrag a or Seqff method. Results are stored in the directory <code class="docutils literal notranslate"><span class="pre">tables</span></code>. Read length distribution are stored in directory <code class="docutils literal notranslate"><span class="pre">read_len_dist</span></code>.</p>
<p>Arguments:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>-r18, --fetalRange_chr18, default=“121:165”, to set the range of reads length to be considered as most likely fetal for chr18
-r21, --fetalRange_chr21, default==“109:164”, to set the range of reads length to be considered as most likely fetal for chr18
</pre></div>
</div>
</div>
<div class="section" id="sao">
<h1>Sao<a class="headerlink" href="#sao" title="Permalink to this headline">¶</a></h1>
<p>Copy number alteration prediction with WisecondorX.</p>
<p>Input list of samples, only prefix file name (no path). Reference files will be automatically found in <code class="docutils literal notranslate"><span class="pre">reference/npz</span></code> as determined by <code class="docutils literal notranslate"><span class="pre">larissa.py</span></code>.</p>
<p>Output: one table with sample names and associated Z-score for all chromosomes estimated by WisecondorX method. Results are stored in the directory <code class="docutils literal notranslate"><span class="pre">tables</span></code>. Bin visualization and additional outputs by WisecondorX are stored in the directory <cite>copy_number_alteration_data`</cite>.</p>
<p>Arguments: none</p>
</div>
<div class="section" id="thalassa">
<h1>Thalassa<a class="headerlink" href="#thalassa" title="Permalink to this headline">¶</a></h1>
<p>Output preparation.</p>
<p>It collects results from upstream modules in a table.</p>
<p>Input: Automatically finds all tables stored in the directory <code class="docutils literal notranslate"><span class="pre">tables</span></code>.</p>
<p>Output: A table in which each sample (line) is described by the following columns: quality control (visual output), sd, gender prediction (magicY), ff (2 columns one for each method), e value for chr18 (2 columns one for each ff method), e value for chr21 (2 columns one for each ff method),  and the Z-score for chromosome 13, 18 and 21.</p>
<p>Arguments: none</p>
</div>


           </div>
           
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, David Pratella

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>